src/main/java/org/truffleruby/parser/BodyTranslator.java

```diff
@@ -2050,6 +2051,10 @@ public RubyNode visitMatch2Node(Match2ParseNode node) {
             final Regex regex = new Regex(bytes, 0, bytes.length, regexpNode.getOptions().toOptions(), regexpNode.getEncoding(), Syntax.RUBY);
 
             if (regex.numberOfNames() > 0) {
+                final ArrayList<RubyNode> setters = new ArrayList<>();
+                final ArrayList<RubyNode> nilSetters = new ArrayList<>();
```

Comment:
I think you could use `regex.numberOfNames()` here so to allocate fixed-size arrays directly.